providers/implementations/macs/kmac_prov.c

```diff
@@ -246,15 +252,15 @@ static int kmac_setkey(struct kmac_data_st *kctx, const unsigned char *key,
     const EVP_MD *digest = ossl_prov_digest_md(&kctx->digest);
     int w = EVP_MD_block_size(digest);
 
-    if (keylen < 4 || keylen > KMAC_MAX_KEY) {
+    if (keylen < KMAC_MIN_KEY || keylen > KMAC_MAX_KEY) {
         ERR_raise(ERR_LIB_PROV, PROV_R_INVALID_KEY_LENGTH);
         return 0;
     }
     if (w < 0) {
         ERR_raise(ERR_LIB_PROV, PROV_R_INVALID_DIGEST_LENGTH);
         return 0;
     }
-    if (!kmac_bytepad_encode_key(kctx->key, &kctx->key_len,
+    if (!kmac_bytepad_encode_key(kctx->key, sizeof(kctx->key), &kctx->key_len,
```

Comment:
you're assuming the `sizeof` here, and below, have no padding.

Reply:
Not quite sure what you are saying.
I am assuming sizeof(unsigned char) is 1., and that sizeof() returns the number of elements in the array that can be used. This is used quit a lot in the code base.

Reply:
NOTE key is this: unsigned char key[KMAC_MAX_KEY_ENCODED];